Sources/CollectionsDifferentiable/Collections.swift

```diff
@@ -0,0 +1,4 @@
+// A module that re-exports Swift Collections API and the Differentiable extensions to the Swift Collections API
+@_exported import Collections
+// Swift Collection modules with Differentiable extensions
+@_exported import OrderedCollectionsDifferentiable
```

Comment:
shouldn't this be in the `OrderedCollectionsDifferetiable` target and this one exports `CollectionsDifferetiable`?

Reply:
Same as below only subtlety here is that we also export all the Differentiable extensions that we write in this repo. For now we'll only have extensions for OrderedCollections.